src/transformers/models/dab_detr/modeling_dab_detr.py

```diff
@@ -0,0 +1,2330 @@
+        self.cross_attn_key_pos_proj = nn.Linear(d_model, d_model)
+        self.cross_attn_value_proj = nn.Linear(d_model, d_model)
+        self.cross_attn_query_pos_sine_proj = nn.Linear(d_model, d_model)
+
+        self.cross_attn = DABDETRAttention(config, is_cross=True)
+        self.decoder_attention_heads = config.decoder_attention_heads
+        self.do_use_self_attn_decoder = config.do_use_self_attn_decoder
+
+        # FFN
+        self.cross_attn_layer_norm = nn.LayerNorm(d_model)
+        self.final_layer_norm = nn.LayerNorm(d_model)
+        self.fc1 = nn.Linear(d_model, config.decoder_ffn_dim)
+        self.fc2 = nn.Linear(config.decoder_ffn_dim, d_model)
+        self.activation_fn = ACT2FN[config.activation_function]
+        self.activation_dropout = config.activation_dropout
+        self.keep_query_pos = config.keep_query_pos
+
+        if not config.keep_query_pos and not is_first:
+            self.cross_attn_query_pos_proj = None
+
+        self.is_first = is_first
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        object_queries: Optional[torch.Tensor] = None,
+        query_position_embeddings: Optional[torch.Tensor] = None,
+        query_sine_embed: Optional[torch.Tensor] = None,
+        encoder_hidden_states: Optional[torch.Tensor] = None,
+        encoder_attention_mask: Optional[torch.Tensor] = None,
+        output_attentions: Optional[bool] = None,
+    ):
+        """
+        Args:
+            hidden_states (`torch.FloatTensor`): input to the layer of shape `(seq_len, batch, embed_dim)`
+            attention_mask (`torch.FloatTensor`): attention mask of size
+                `(batch, 1, target_len, source_len)` where padding elements are indicated by very large negative
+                values.
+            object_queries (`torch.FloatTensor`, *optional*):
+                object_queries that are added to the queries and keys
+            in the cross-attention layer.
+            query_position_embeddings (`torch.FloatTensor`, *optional*):
+                object_queries that are added to the queries and keys
+            in the self-attention layer.
+            encoder_hidden_states (`torch.FloatTensor`):
+                cross attention input to the layer of shape `(seq_len, batch, embed_dim)`
+            encoder_attention_mask (`torch.FloatTensor`): encoder attention mask of size
+                `(batch, 1, target_len, source_len)` where padding elements are indicated by very large negative
+                values.
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+
+        """
+        residual = hidden_states
+
+        # ========== Begin of Self-Attention =============
+        if self.do_use_self_attn_decoder:
```

Comment:
If you want to train a model from scratch instead of fine-tuning, you can set that parameter to False and use the model accordingly, while utilizing the decoder part without the self-attention layer. However, there isn't a pre-trained version of such a model, as the pre-trained models always have this parameter set to True.